### PR TITLE
Refactor/#12 remove cli

### LINKED ISSUE #12: Remove CLI stuff
Remove cli prompts and console.logs

### PR DESCRIPTION
**Pull request by asbjornh**
Fixes #12 

Removes all logging and prompt.
- The functions now expect to get all required information handed to them.
  - The functions will not try to resolve .creunarc.json or .eslintrc.json anymore
  - eslintrc.json is now optional
- Functions expect an options object instead of arguments
- All functions now returns a promise that's either rejected with an error message or resolved with a list of confirmation messages. Not all of the functions are actually async, but I think having consistent interfaces is good anyway.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,16 +13,16 @@ This module exports one object containing all the functions.
...
+    rename, // Rename component
+    toStateful, // Convert component to stateful
+    toStateless; // Convert component to stateless
 }
 ```
 
 ## Notes
 
-The functions require a `.creunarc.json` and a `.eslintrc.json` in your project root.
+When running from `@creuna/cli` the functions require a `.creunarc.json` in your project root.
```
Comment (pellebjerkestrand) [inline README.md]: Just refer to the `@creuna/cli` documentation. The requirement should be documented there.

File: source/new-component.js [hunk 2]
```diff
@@ -14,63 +13,70 @@ const readFile = require('./utils/read-file');
...
-      type: Boolean,
-      value: maybeShouldBeStateful
+module.exports = function({
+  componentsPath,
+  eslintConfig,
+  pathOrName,
+  shouldBeStateful
+}) {
+  return new Promise(async (resolve, reject) => {
+    if (!componentsPath) {
```
Comment (pellebjerkestrand) [inline source/new-component.js]: Use `ow` for this? (https://github.com/sindresorhus/ow)

Comment (pellebjerkestrand) [reply source/new-component.js]: This is the case for everywhere we use a check and reject pattern.

File: source/utils/get-component.js [hunk 3]
```diff
@@ -53,6 +25,19 @@ function parse({ pathOrName, componentsPath }) {
     });
   }
 
+  // Handle file in current working directory
+  if (hasFileExtension) {
+    return validatePath({
```
Comment (pellebjerkestrand) [inline source/utils/get-component.js]: `isPathValid` to conform to the rest of the functions.

Comment (pellebjerkestrand) [reply source/utils/get-component.js]: No, that doesn't make sense when I look at the implementation. The name is nevertheless confusing for me.

File: source/new-component.js (hunk 2 above)
Comment (asbjornh) [reply source/new-component.js]: Do we _really_ need to add a 240kb dependency to save ~15 lines?

File: README.md (hunk 1 above)
Comment (asbjornh) [reply README.md]: Agreed

File: source/utils/get-component.js (hunk 3 above)
Comment (asbjornh) [reply source/utils/get-component.js]: Fixed

**Pull request merged**